src/tree/gpu_hist/histogram.cuh

```diff
@@ -17,12 +17,26 @@ DEV_INLINE T TruncateWithRoundingFactor(T const rounding_factor, float const x)
   return (rounding_factor + static_cast<T>(x)) - rounding_factor;
 }
 
+struct HistogramLaunchConfig {
+  unsigned grid_size;
+  unsigned block_threads;
+  bool shared;
+  HistogramLaunchConfig(unsigned grid_size, unsigned block_threads, bool shared) :
+    grid_size(grid_size), block_threads(block_threads), shared(shared) {}
+  HistogramLaunchConfig() : grid_size(0), block_threads(0), shared(false) {}
```

Comment:
What happens if this gets used with default parameters?

Reply:
No longer applies.